docs/effect-package.md

```diff
@@ -0,0 +1,34 @@
+# `effect` package
```

Comment:
Should this just be in the actual package itself; inside a `package.go` file? Similar to https://github.com/temporalio/temporal/blob/main/internal/goro/package.go

Reply:
I believe we agreed that all actual docs should be in `docs` root dir. Only `README.md` are allowed in specific dirs and they should have only link to file in `docs`. I will add `README.md` there.

Reply:
We did say that 👍 at the same time, I think of the `docs/` as high-level, conceptual documentation. This one feels more like a low-level, code comment (with some extra context).

I have no objections to leaving it here.